common/walker/src/report.rs

```diff
@@ -124,8 +127,77 @@ impl SplitScannerError for Result<Report, ScannerError> {
 }
 
 /// Handle the report
-pub async fn handle_report(report: Report) -> anyhow::Result<()> {
-    // FIXME: this is a very simplistic version of handling the error
-    log::info!("Import report: {report:#?}");
+pub async fn handle_report(report: Report, report_type: String) -> anyhow::Result<()> {
+    let path = env::var_os("REPORT_PATH").unwrap_or_else(|| OsString::from(DEFAULT_REPORT_OUTPUT_PATH));
+    let path = path.to_str().unwrap_or(DEFAULT_REPORT_OUTPUT_PATH).to_string();
+
+    let template_content = include_str!("../templates/report.html");
+    let mut tera = Tera::default();
+    tera.add_raw_template("report.html", template_content)?;
+
+    let mut context = Context::new();
+    let current_time = OffsetDateTime::now_utc();
+    context.insert("report", &report);
+    context.insert("type", &report_type);
+    context.insert("current_time", &current_time);
+
+    let format = format_description!("[year]-[month]-[day]--[hour]-[minute]-[second]");
+    let formatted_time = current_time.format(&format)?;
+    match tera.render("report.html", &context) {
+        Ok(rendered_html) => {
+            let out_put_html_path = format!("{path}/{report_type}/html/report-{formatted_time}.html");
+            if let Some(parent) = std::path::Path::new(&out_put_html_path).parent() {
+                fs::create_dir_all(parent)?;
+            }
+            fs::write(out_put_html_path.clone(), rendered_html)?;
+            log::info!("Successfully generated the report file.");
+        }
+        Err(e) => {
```

Comment:
If the report fails to write, we raise an error. If we can't create the parent directory, we raise an error. If we can't format the timestamp, we raise an error, … why is not being able to render the report being ignored as an error?

Reply:
it's means, we just needs to throw them, do not catch one of them. is it right?

Reply:
I think that would be consistent.